examples/task_controller_client/section_control_implement_sim.cpp

```diff
@@ -0,0 +1,318 @@
+			{
+				value = 3785000; // 1000 Gal in ml;
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SectionControlState):
+			{
+				value = 1; // AUTO mode
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::PrescriptionControlState):
+			{
+				value = 1; // AUTO mode
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState1_16):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState17_32):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState33_48):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState49_64):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState65_80):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState81_96):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState97_112):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState113_128):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState129_144):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState145_160):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState161_176):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState177_192):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState193_208):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState209_224):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState225_240):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState241_256):
+			{
+				constexpr std::uint8_t NUMBER_SECTIONS_PER_CONDENSED_MESSAGE = 16;
+				std::uint16_t sectionIndexOffset = NUMBER_SECTIONS_PER_CONDENSED_MESSAGE * (DDI - static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState1_16));
+				value = 0;
+
+				for (std::uint_fast8_t i = 0; i < NUMBER_SECTIONS_PER_CONDENSED_MESSAGE; i++)
+				{
+					value |= ((true == sim->get_section_state(i + sectionIndexOffset)) ? static_cast<std::uint32_t>(0x01) : static_cast<std::uint32_t>(0x00)) << (2 * i);
+				}
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualVolumePerAreaApplicationRate):
+			{
+				value = sim->get_actual_rate();
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualWorkState):
+			{
+				value = sim->get_actual_work_state();
+			}
+			break;
+
+			default:
+			{
+				value = 0;
+			}
+			break;
+		}
+	}
+	return true; // You would want to return false in a real controller if a value could not be retrieved
```

Comment:
Kind of curious here? Why can't we return false in our example? How would a real controller be different from this one except for returning real values?

Reply:
Oh, it's just shorthand for returning 0 for values handled by the switch's default statement. The actual use of the return type is for the TC client to know if it needs to keep calling more callbacks to search for one that can satisfy the element number + DDI combination it needs. But in the example this is the only value command callback, so we always want to return true. Maybe that isn't clear enough... 😣

Reply:
Clarified the comments to hopefully make it more clear to a reader what's happening there.